lib/exercises.rb

```diff
@@ -1,29 +1,112 @@
-
 # This method will return an array of arrays.
 # Each subarray will have strings which are anagrams of each other
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: o(n2) because I am iterating over the string and
+# I know Bytes is gonna iterate over it as well
+# Space Complexity: 0(n)
 
 def grouped_anagrams(strings)
-  raise NotImplementedError, "Method hasn't been implemented yet!"
+  hash_ana = {}
+  outerArray = []
+
+  # The way I identify they are anagrams is using bytes to get the ASCII value of each char sum them,
+  # Then we know all the characters have the same letter on it.
+  strings.each do |string|
+    if hash_ana[string.bytes.sum]
+      hash_ana[string.bytes.sum].push(string)
+    else
+      hash_ana[string.bytes.sum] = [string]
+    end
+  end
+
+  hash_ana.each do |key|
+    outerArray << key[1]
+  end
+
+  return outerArray
 end
 
 # This method will return the k most common elements
 # in the case of a tie it will select the first occuring element.
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: o(n)
+# Space Complexity: o(1)
 def top_k_frequent_elements(list, k)
```

Comment:
You're sorting here, so this isn't O(n) in time complexity.  You are also building a hash so you have an O(n) space complexity.